### PR TITLE
[New JS Importer] Discuss canonicalizing relative URLs in the summary

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: proposal/new-js-importer.d.ts [hunk 1]
```diff
@@ -82,6 +83,32 @@
  * Having a canonical URL for each stylesheet allows Sass to ensure that the
  * same stylesheet isn't loaded multiple times in the new module system.
  *
+ * #### Canonicalizing Relative Loads
+ *
+ * When a stylesheet tries to load a relative URL, such as `@use "variables"`,
+ * it's not clear from the document itself text whether that's meant to refer to
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts]:
```suggestion
 * it's not clear from the document itself whether that refers to
```

File: proposal/new-js-importer.d.ts [hunk 2]
```diff
@@ -82,6 +83,32 @@
...
+ * a file that exists relative to the stylesheet or to another importer or load
+ * path. Here's how the importer API resolves that ambiguity:
+ *
+ * * First, the relative URL is resolved relative to the old stylesheet's
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts]: Might help to clarify the meaning of `old stylesheet`

File: proposal/new-js-importer.d.ts [hunk 3]
```diff
@@ -82,6 +83,32 @@
...
+ *   canonical URL. For example, if the canonical URL is
+ *   `file:///path/to/my/_styles.scss`, then the resolved URL will be
+ *   `file://path/to/my/variables`.
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts]: Should this be
```suggestion
 *   `file:///path/to/my/variables`.
```
?

File: proposal/new-js-importer.d.ts [hunk 4]
```diff
@@ -82,6 +83,32 @@
...
+ *
+ * * This URL is then passed to the `canonicalize()` method of the importer that
+ *   loaded the old stylesheet. (That means it's important for your importers to
+ *   support absolute URLs!) If the importer recognizes it, it returns the
+ *   canonical value which is then passed to `load()`; otherwise, it returns
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts]: Might help to be explicit about whose `load()` this is

File: proposal/new-js-importer.d.ts [hunk 5]
```diff
@@ -82,6 +83,32 @@
...
+ *   `null`.
+ *
+ * * If the old stylesheet's importer didn't recognize the URL, it's passed to
+ *   all the `importers`' canonicalize functions in order, then checked for in
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts]: The `in order` modifier is a bit ambiguous.

File: proposal/new-js-importer.d.ts [hunk 6]
```diff
@@ -82,6 +83,32 @@
...
+ *   all the `loadPaths`. If none of those recognizes it, the load fails.
+ *
+ * It's important that local relative paths take precedence over other importers
+ * or load paths, because otherwise your local stylesheets could get
+ * unexpectedly broken by a dependency adding a file with a conflicting name.
```
Comment (Awjin) [inline proposal/new-js-importer.d.ts:111]: This section is a bit vague. A toy example could help motivate/explain it

**Review by Awjin — CHANGES_REQUESTED** [review]

File: proposal/new-js-importer.d.ts (hunk 6 above)
Comment (nex3) [reply proposal/new-js-importer.d.ts:111]: This is already a bit of a tangent on a tangent, so I'd rather not add too much additional detail.

**Review by Awjin — APPROVED** [review]

**Pull request merged**